### ISSUE TITLE
Issue with specific URL

### ISSUE THREAD

**Issue by alectrocute** [ospfranco/link-preview-js#95]
Using this URL causes `getLinkPreview` to never return:

```
https://www.gamestop.com/video-games/pc-gaming/components/cooling/products/hyper-212-rgb-black-edition-fan/185243.html?gclid=Cj0KCQjwraqHBhDsARIsAKuGZeECDlqkF2cxpcuS0xRxQmrv5BxFawWS_B51kiqehPf64_KlO0oyunsaAhn5EALw_wcB&gclsrc=aw.ds
```

Found this while investigating why a server I had running was crashing. Even if the link isn't supported, shouldn't `getLinkPreview` at least throw an error instead of hang infinitely?

**Comment by ospfranco** [comment]
The internal implementation does not have any timeout mechanisms built in, but this is a good suggestion, if I find some time I will add it, but PRs are welcome

**Comment by ospfranco** [comment]
Documentation for a solution:

https://dmitripavlutin.com/timeout-fetch-request/

**Comment by ospfranco** [comment]
Implemented/released in 2.1.7 as a `timeout` parameter in the options object, the default implementation now also has a timeout of 3 seconds

**Comment by alectrocute** [comment]
@ospfranco You're amazing!!
